embedded/document/engine.go

```diff
@@ -716,8 +716,8 @@ func (e *Engine) upsertDocuments(ctx context.Context, sqlTx *sql.SQLTx, collecti
 
 		provisionedDocID, docIDProvisioned := doc.Fields[docIDFieldName]
 		if docIDProvisioned {
-			if isInsert {
-				return 0, nil, fmt.Errorf("%w: field (%s) should NOT be specified when inserting a document", ErrIllegalArguments, docIDFieldName)
+			if isInsert && docIDFieldName == DefaultDocumentIDField {
```

Comment:
if I read correctly, if inserted document does not have custom id, then immudb will autogenerate one. I would block that, as it is doing something by the user behind the scenes, so something that I would not expect when the user explicitly specifies that document id is provided externally. I.e. even if ids are just hex encoded strings at the end, they might have more meaning from the user perspective.

Reply:
Indeed. In case of missing custom ID provided by the user it will be auto-generated. This is similar strategy for example Elasticsearch uses. If user doesn't specify an identifier it is automatically generated. Each approach has pros and cons of course. In https://github.com/codenotary/immudb/issues/1755 I mentioned that maybe it's not worth giving a user flexibility to provide document DB with custom ID field just force the name of ID field but let user customize the value. In such a case it's even more important to have auto-generated `_id` if not provided or use the one from the user if available.